add `Oscar.versioninfo()` function and short technical description of oscar architecture (#1122)
oscar-system/Oscar.jl: docs/src/General/architecture.md

```@contents
Pages = ["architecture.md"]
```

```@meta
CurrentModule = Oscar
```

```@setup oscar
using Oscar
```

# Architecture

This page aims to give a short technical overview of the architecture of Oscar.
A more in-depth overview of the various components of Oscar is given
[on the Oscar homepage](https://oscar.computeralgebra.de/about/).

## Julia packages

Oscar is developed as a pure julia package Oscar.jl and builds on the features
and interfaces provided by the julia packages for the cornerstones:
  - ANTIC: [AbstractAlgebra.jl](https://github.com/Nemocas/AbstractAlgebra.jl),
           [Hecke.jl](https://github.com/thofma/Hecke.jl),
           [Nemo.jl](https://github.com/nemocas/Nemo.jl)
  - [GAP.jl](https://github.com/oscar-system/GAP.jl)
  - [Polymake.jl](https://github.com/oscar-system/Polymake.jl)
  - [Singular.jl](https://github.com/oscar-system/Singular.jl)

The packages are integrated into the julia package manager and will be

> installed automatically together with Oscar and can also be accessed directly

Reviewer: ```suggestion
installed automatically as dependencies of Oscar. They can be accessed directly
```
What do you think about this?
